src/Sylius/Component/Core/spec/Uploader/ImageUploaderSpec.php

```diff
@@ -46,8 +46,25 @@ function it_uploads_an_image($filesystem, $image)
         $image->hasFile()->willReturn(true);
         $image->getPath()->willReturn('foo.jpg');
 
-        $filesystem->delete(Argument::any())->shouldBeCalled();
-        $filesystem->write(Argument::any(), Argument::any())->shouldBeCalled();
+        $filesystem->has('foo.jpg')->willReturn(false);
+
+        $filesystem->delete(Argument::any())->shouldNotBeCalled();
+        $filesystem->write('foo.jpg', Argument::any())->shouldBeCalled();
+
+        $image->setPath(Argument::any())->shouldBeCalled();
+
+        $this->upload($image);
+    }
+
+    function it_replaces_an_image($filesystem, $image)
+    {
+        $image->hasFile()->willReturn(true);
+        $image->getPath()->willReturn('foo.jpg');
```

Comment:
Naming both files `foo.jpg` is misleading, let's name the currently uploaded one `bar.jpg` and the existing one `foo.jpg`?